include/nlohmann/detail/macro_scope.hpp

```diff
@@ -230,6 +230,46 @@
         e = ((it != std::end(m)) ? it : std::begin(m))->first;                                  \
     }
 
+#if (defined(__cpp_exceptions) || defined(__EXCEPTIONS) || defined(_CPPUNWIND)) && !defined(JSON_NOEXCEPTION)
+/*!
+@brief macro to briefly define a mapping between an enum and JSON
+@def NLOHMANN_JSON_SERIALIZE_ENUM
+@since version 3.4.0
+*/
+#define NLOHMANN_JSON_SERIALIZE_ENUM_STRICT(ENUM_TYPE, ...)                                                                   \
+    template<typename BasicJsonType>                                                                                          \
+    inline void to_json(BasicJsonType& j, const ENUM_TYPE& e)                                                                 \
+    {                                                                                                                         \
+        static_assert(std::is_enum<ENUM_TYPE>::value, #ENUM_TYPE " must be an enum!");                                        \
+        static const std::pair<ENUM_TYPE, BasicJsonType> m[] = __VA_ARGS__;                                                   \
+        auto it = std::find_if(std::begin(m), std::end(m), [e](const std::pair<ENUM_TYPE, BasicJsonType>& ej_pair) -> bool {  \
+            return ej_pair.first == e;                                                                                        \
+        });                                                                                                                   \
+        if (it == std::end(m))                                                                                                \
+        {                                                                                                                     \
+            throw nlohmann::detail::type_error::create(302,                                                                   \
+                    nlohmann::detail::concat("can't serialize enum value ", std::to_string(static_cast<int>(e)),              \
```

Comment:
Would it make sense to say something about the value not being a enum value listed in `NLOHMANN_JSON_SERIALIZE_ENUM_STRICT`?

Reply:
yes, this is a great suggestion.